CMakeLists.txt

```diff
@@ -316,17 +316,26 @@ set_var_to_condition(MIOPEN_USE_MLIR_DEFAULT NOT (NOT ${BUILD_SHARED_LIBS} AND $
 option(MIOPEN_USE_MLIR "Use MLIR compilation backend" ${MIOPEN_USE_MLIR_DEFAULT})
 
 if(MIOPEN_USE_MLIR)
-    # REQUIRED is not supported before cmake 3.18
-    find_library(LIBMLIRMIOPEN MLIRMIOpen REQUIRED)
     if(NOT ${BUILD_SHARED_LIBS} AND ${MIOPEN_USE_COMGR})
-        message(FATAL_ERROR "Potential symbol conflict between mlir and comgr in static build")
+      message(FATAL_ERROR "Potential symbol conflict between mlir and comgr in static build")
     endif()
-
-    if(NOT LIBMLIRMIOPEN)
-        message(FATAL_ERROR "libMLIRMIOpen not found, please reinstall dependencies. \
+    # Try to find package Miir
+    # REQUIRE is omitted since we do not want cmake to abort if the package is not found
+    find_package(Miir 1.0.0 CONFIG)
+    if(NOT Miir_FOUND)
+      message(STATUS "Falling back to find library libMLIRMIOpen")
+      # Backward compatibility
```

Comment:
Please indicate the max ROCm version this piece of code maintains backward compatibility with. This would help us clean up the code in the future. For example, if this is 5.3, then we will be able to remove this piece of code as soon 5.3 support is dropped.

Reply:
Note: because of the backward compatibility requirement I'd prefer the package name to be renamed to the correct one instead of `Miir` right now. (If we do it in next release we'd have to carry this package name in future for backward compatibility purpose again)

Reply:
@jerryyin Very good!